### ISSUE TITLE
Docs: Add basic usage example to all components

### ISSUE THREAD

**Issue by benelan** [Esri/calcite-components#2860]
### Description 
There are a lot of components without usage examples. I will add at least a 'Basic' example to all of them.

**Comment by macandcheese** [comment]
We'd like to eventually consume these samples in the developer site, where we can generate component samples from the provided snippets.

It would be great if each example could have explicit data points for title, description, and then the code sample. Perhaps having individual files for each sample?

Right now, it's hard to parse the returned files since it's often a mix of inline code, text, other markdown formatting, etc., cc @caripizza

**Comment by benelan** [comment]
@macandcheese Do you want to set up a meeting next week to discuss this? I can do it however works best for you and then go through and change the rest of them too.

**Comment by benelan** [comment]
Components that still have missing usage examples:
- [ ] card
- [ ] input-date-picker
- [ ] input-time-picker
- [ ] calcite-radio-group
- [ ] calcite-rating (needs another)
- [ ] link
- [ ] pagination (needs rework)
- [ ] progress (has a TODO header??)
- [ ] tile

**Comment by benelan** [comment]
Installed, @julio8a or @jcfranco, can I get a verification please.

**Comment by julio8a** [comment]
Verified on master
